### ISSUE TITLE
Can't deploy contracts to dev chain in 2022.06.06 and 06.04

### ISSUE THREAD

**Issue by mohoff** [ledgerwatch/erigon#4485]
#### System information

Erigon version: `2022.06.04`

Commit hash : `e866889b60f8854d574c54686792488a2cacba95`

#### Expected behaviour

Dev chain smart contract deployments succeed

#### Actual behaviour

First contract deployment fails with `header for hash not found`. The erigon logs show nothing special. RPC daemon logs are attached. The same series of contract deployements worked under `v2022.06.02`.

```
[TRACE] [06-18|02:04:44.139] Served                                   conn=172.20.0.6:41804 t=12.887972ms method=eth_getBlockByNumber reqid=5735023419010213 params="[\"0x18\",false]"
[TRACE] [06-18|02:04:44.139] Served                                   conn=172.20.0.6:41790 t=15.509286ms method=eth_getBlockByNumber reqid=91063108417855 params="[\"0x18\",false]"
[WARN] [06-18|02:04:44.152] Served                                   conn=172.20.0.6:41824 method=eth_estimateGas reqid=10 t=883.852µs err="header for hash not found"
[TRACE] [06-18|02:04:44.152] Served                                   conn=172.20.0.6:41824 t=920.82µs method=eth_estimateGas reqid=10 params="[{\"from\":\"0xf0ddad96597e10427cba825d7e6cb327fc57f6d3\",\"data\":\"0x....snip\"}]"
[TRACE] [06-18|02:04:44.162] Served                                   conn=172.20.0.6:41826 t=3.597001ms method=eth_gasPrice reqid=11 params=[]
[TRACE] [06-18|02:04:44.170] Served                                   conn=172.20.0.6:41828 t=2.116799ms method=eth_getTransactionCount reqid=5260174556562551 params="[\"0xf0ddad96597e10427cba825d7e6cb327fc57f6d3\",\"pending\"]"
[WARN] [06-18|02:04:44.172] Served                                   conn=172.20.0.6:41830 method=eth_estimateGas reqid=4426342974886299 t=817.604µs err="header for hash not found"
[TRACE] [06-18|02:04:44.172] Served                                   conn=172.20.0.6:41830 t=919.657µs method=eth_estimateGas reqid=4426342974886299 params="[{\"from\":\"0xf0ddad96597e10427cba825d7e6cb327fc57f6d3\",\"data\":\"0x....snip\",\"gasPrice\":\"0x26b084b\"}]"
[TRACE] [06-18|02:04:44.183] Served                                   conn=172.20.0.6:41832 t=1.335959ms method=eth_call reqid=1655517884178 params="[{\"from\":\"0xf0ddad96597e10427cba825d7e6cb327fc57f6d3\",\"data\":\"0x....snip\",\"gasPrice\":\"0x26b084b\"},\"latest\"]"
[TRACE] [06-18|02:04:51.019] Served                                   conn=172.20.0.8:33558 t=1.35894ms method=eth_getTransactionCount reqid=1 params="[\"0x95c2d93dc7e29883e02c5e8c79856f9af99109fe\",\"pending\"]"
[TRACE] [06-18|02:04:51.036] Served                                   conn=172.20.0.8:33560 t=1.332407ms method=eth_getTransactionCount reqid=2 params="[\"0x95c2d93dc7e29883e02c5e8c79856f9af99109fe\",\"pending\"]"
[TRACE] [06-18|02:04:51.170] Served                                   conn=172.20.0.8:33564 t=740.429µs method=eth_getBlockByNumber reqid=3 params="[\"pending\",false]"
[TRACE] [06-18|02:04:51.180] Served                                   conn=172.20.0.8:33566 t=375.42µs method=eth_getBalance reqid=4 params="[\"0x95c2d93dc7e29883e02c5e8c79856f9af99109fe\",\"pending\"]"
[INFO] [06-18|02:04:51.466] Submitted transaction                    hash=0xf9a6ca755dda020c85234d401e2dabb2b5136d39b9be05487713a08f50e928d8 from=0x95C2d93DC7E29883e02C5e8c79856F9af99109FE nonce=0 recipient=0x95709Dd278d3BbC997eEe7Ab8dcf6D77aB08E545 value=0x0
[TRACE] [06-18|02:04:51.466] Served                                   conn=172.20.0.8:33570 t=3.689992ms method=eth_sendRawTransaction reqid=5 params="[\"0x...snip\"]"
[WARN] [06-18|02:04:51.471] Served                                   conn=172.20.0.8:33572 method=eth_sendRawTransaction reqid=6 t=999.032µs err="ALREADY_EXISTS: already known"
```

**Comment by mohoff** [comment]
Facing the same on latest release `2022.06.06`, commit is `588c2d4e36b68a59580154fc20f4c3e75ad36623`

**Comment by mandrigin** [comment]
@mohoff can you provide the full transaction so I can debug locally?

**Comment by dmitriyselivanov** [comment]
@mohoff @mandrigin the issue was fixed in [d72fba3](https://github.com/ledgerwatch/erigon/commit/d72fba3bec6beb4ef829f626760ff3725152a6f1)
I am not facing the issue at the current devel branch commit [8a75033](https://github.com/ledgerwatch/erigon/commit/8a75033b9860aa019f4ef88280f52f1abbfe8536) as well.
